library/src/main/java/com/paypal/messages/logger/Logger.kt

```diff
@@ -54,9 +58,9 @@ class Logger private constructor() {
 			partnerAttributionId = null,
 			// merchantProfileHash will be later defined by pulling from LocalStorage
 			merchantProfileHash = null,
-			deviceId = Settings.Secure.ANDROID_ID,
+			deviceId = if (deviceId == "") Settings.Secure.ANDROID_ID else deviceId,
```

Comment:
I believe the setting.secure is used like this `Settings.Secure.getString(context.contentResolver, Settings.Secure.ANDROID_ID)`

I'm seeing android_id returned with just Settings.Secure.ANDROID_ID